cloudinit/net/ephemeral.py

```diff
@@ -417,21 +421,32 @@ def get_first_option_value(
 class EphemeralIPNetwork:
     """Marries together IPv4 and IPv6 ephemeral context managers"""
 
-    def __init__(self, interface, ipv6: bool = False, ipv4: bool = True):
+    def __init__(
+        self,
+        interface,
+        ipv6: bool = False,
+        ipv4: bool = True,
+        alt_tmp_dir=None,
+    ):
         self.interface = interface
         self.ipv4 = ipv4
         self.ipv6 = ipv6
         self.stack = contextlib.ExitStack()
         self.state_msg: str = ""
+        self.alt_tmp_dir = alt_tmp_dir
 
     def __enter__(self):
         # ipv6 dualstack might succeed when dhcp4 fails
         # therefore catch exception unless only v4 is used
         try:
             if self.ipv4:
-                self.stack.enter_context(EphemeralIPv6Network(self.interface))
+                self.stack.enter_context(
```

Comment:
Fixing the reversed ipv4/ipv6 context managers should probably be a separate PR. I suspect this technically breaks ephemeral ipv4 on ec2 currently (which is not currently in any releases).

Reply:
Extracted here: https://github.com/canonical/cloud-init/pull/1697